# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxUserFontSet.h
# Commit: 4f9d4d1aaa3e
# Full Hash: 4f9d4d1aaa3e4875ad481fac5c503cd61af94a88
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2021-04-07 03:19:44
# Regressor Bug: 1694123
# File Overlap Count: 1
# Description:
#   Bug 1694123 - Ensure user fonts with src:local() are refreshed after a font-list update, even if the local font lookup previously failed. r=jwatt
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D106511
# ==============================================================================

diff -r 43764851d067 -r 4f9d4d1aaa3e gfx/thebes/gfxUserFontSet.h
--- a/gfx/thebes/gfxUserFontSet.h	Tue Apr 06 16:23:17 2021 +0000
+++ b/gfx/thebes/gfxUserFontSet.h	Tue Apr 06 16:23:17 2021 +0000
@@ -618,6 +618,9 @@
     mUserFontLoadState = STATUS_NOT_LOADED;
     mFontDataLoadingState = NOT_LOADING;
     mLoader = nullptr;
+    // Reset mCurrentSrcIndex so that all potential sources are re-considered.
+    mCurrentSrcIndex = 0;
+    mSeenLocalSource = false;
   }
 
   // whether to wait before using fallback font or not
@@ -777,6 +780,7 @@
   };
   FontDataLoadingState mFontDataLoadingState;
 
+  bool mSeenLocalSource;
   bool mUnsupportedFormat;
   mozilla::StyleFontDisplay mFontDisplay;  // timing of userfont fallback
 
